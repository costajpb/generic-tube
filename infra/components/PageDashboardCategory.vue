<script setup lang="ts">
import type Dashboard from '@/application/dashboard'
import type Show from '@/domain/show/entity'
import { inject } from 'vue'
import PageDashboardCategoryShowSkeleton from '@/infra/components/PageDashboardCategoryShowSkeleton.vue'

const props = defineProps<{
  name: string
  shows: Show[]
}>()

const useCase = inject('useCase') as Dashboard

const scrollHorizontally = (event: WheelEvent) => {
  const offset = event.deltaY
  ;(event.currentTarget as HTMLElement).scrollLeft += offset
}
</script>

<template>
  <article :class="classes.category">
    <h2 :class="classes.title">{{ props.name }}</h2>
    <ol :class="classes.shows" @wheel.prevent="scrollHorizontally">
      <PageDashboardCategoryShowSkeleton v-for="show in props.shows" :key="show.id">
        <a
          :class="classes.anchor"
          :href="'/shows/' + show.id"
          @click.prevent="useCase.display(show)"
        >
          <img :class="classes.cover" :src="show.coverImage" :alt="show.title" />
        </a>
      </PageDashboardCategoryShowSkeleton>
    </ol>
  </article>
</template>

<style module="classes" lang="postcss">
.title {
  font-size: var(--font-size-h2);
  color: var(--color-branding-primary-200);
  margin: 0;
}

.shows {
  display: flex;
  overflow: auto;
  margin: 0;
  padding: 0;
  list-style: none;

  > * + * {
    margin-left: var(--size-2);
  }
}

.anchor {
  position: absolute;
  width: 100%;
  height: 100%;
  color: var(--color-text-muted-on-light);
  text-align: center;
}

.cover {
  object-fit: cover;
  width: 100%;
  height: 100%;
  font-style: italic;
}
</style>
