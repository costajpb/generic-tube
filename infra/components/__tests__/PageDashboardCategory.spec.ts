import snapshowWrapper from "@/infra/__tests__/snapshotWrapper"
import PageDashboardCategory from "../PageDashboardCategory.vue"

describe('infra/components/PageDashboardCategory', () => {
    it('should render', () => {
        expect(snapshowWrapper(PageDashboardCategory, {
            provide: {
                useCase: {
                    display: () => {}
                }
            }
        }, {
            name: 'Category',
            shows: []
        })).toMatchSnapshot()
    })
})