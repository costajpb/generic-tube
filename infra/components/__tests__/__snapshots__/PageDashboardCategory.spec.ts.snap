// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`infra/components/PageDashboardCategory > should render 1`] = `
<article
  class="_category_77e56f"
>
  <h2
    class="_title_77e56f"
  />
  <ol
    class="_shows_77e56f"
  >
    
    
  </ol>
</article>
`;
